src/coreclr/gc/gc.cpp

```diff
@@ -1413,7 +1413,7 @@ bool gc_heap::should_move_heap (GCSpinLock* msl)
 
 // All the places where we could be stopped because there was a suspension should call should_move_heap to check if we need to return
 // so we can try another heap or we can continue the allocation on the same heap.
-enter_msl_status gc_heap::enter_spin_lock_msl (GCSpinLock* msl)
+enter_msl_status gc_heap::enter_spin_lock_msl_helper (GCSpinLock* msl)
```

Comment:
can we change this method a bit so we don't immediately need to do another Interlocked check? you could change it to a do/while loop. 

also we should run some microbenchmarks to see if this regresses anything (especially WKS GC)...